File: SOURCE_WEBAPI/VecompSoftware.DocSuiteWeb.Entity/DocumentArchives/DocumentSeriesItemLink.cs
using System;
using VecompSoftware.DocSuiteWeb.Entity.Resolutions;

namespace VecompSoftware.DocSuiteWeb.Entity.DocumentArchives
{

    public class DocumentSeriesItemLink : DSWBaseEntity
    {
        #region [ Constructor ]

        public DocumentSeriesItemLink() : this(Guid.NewGuid()) { }
        public DocumentSeriesItemLink(Guid uniqueId)
            : base(uniqueId)
        {

        }
        #endregion

        #region [ Properties ]   
        public string LinkType { get; set; }

        #endregion

        #region [ Navigation Properties ]
        public virtual DocumentSeriesItem DocumentSeriesItem { get; set; }
        public virtual Resolution Resolution { get; set; }
        #endregion
    }
}

File: SOURCE_DOCSUITE/VecompSoftware.DocSuiteWeb.Data/DocumentSeries/domain/DocumentSeriesItemLink.vb
Imports VecompSoftware.Helpers.ExtensionMethods

<Serializable()> _
Public Class DocumentSeriesItemLink
    Inherits DomainObject(Of Int32)
    Implements IAuditable

#Region " Properties "
    Public Overridable Property DocumentSeriesItem() As DocumentSeriesItem
    Public Overridable Property Resolution As Resolution
    Public Overridable Property LinkType As DocumentSeriesItemLinkType
    Public Overridable Property RegistrationUser As String Implements IAuditable.RegistrationUser
    Public Overridable Property RegistrationDate As DateTimeOffset Implements IAuditable.RegistrationDate
    Public Overridable Property LastChangedUser As String Implements IAuditable.LastChangedUser
    Public Overridable Property LastChangedDate As DateTimeOffset? Implements IAuditable.LastChangedDate
    Public Overridable Property UniqueIdDocumentSeriesItem As Guid
    Public Overridable Property UniqueIdResolution As Guid
#End Region

#Region " Constructor "

    Public Sub New()
        UniqueId = Guid.NewGuid()
    End Sub

#End Region
End Class